Firestore/Swift/Source/Codable/DocumentReference+ReadDecodable.swift

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import FirebaseFirestore
+
+public extension DocumentReference {
+  /// Fetches and decodes the document referenced by this `DocumentReference`.
+  ///
+  /// This allows users to retrieve a Firestore document and have it decoded to an instance of
+  /// caller-specified type as follows:
+  /// ```swift
+  /// ref.getDocument(as: Book.self) { result in
+  ///   do {
+  ///     let book = try result.get()
+  ///   } catch {
+  ///     // Handle error
+  ///   }
+  /// }
+  /// ```
+  /// - Parameters:
+  ///   - as: A `Decodable` type to convert the document fields to.
+  ///   - serverTimestampBehavior: Configures how server timestamps that have
+  ///     not yet been set to their final value are returned from the snapshot.
+  ///   - decoder: The decoder to use to convert the document. Defaults to use
+  ///     default decoder.
+  ///   - completion: The closure to call when the document snapshot has been fetched and decoded.
+  func getDocument<T: Decodable>(as type: T.Type,
```

Comment:
Ideally (but optional):
- Rewrap the comments for the `param` section
- Describe the behavior for nil/offline

Reply:
I'm sorry if I already asked this previously, but I'm afraid I don't remember the answer: do you use a tool to wrap comments?

Reply:
I found `fold` and tried rewrapping - I hope I did it as you envisioned it.
I added offline behavior description from Objective-C method.

I'm not sure what you mean by 'nil' in "Describe the behavior for nil/offline" - could you elaborate?

Reply:
I am actually not sure how the tooling works - ./check/style.sh might wrap comments with lines that are too long, but I don't remember out of the top of my head. I certainly do a lot of manual rewrapping. 

As for describing the method behavior, what about:

```
  /// This method attempts to provide up-to-date data when possible by waiting
  /// for data from the server, but it may return cached data or fail if you are
  /// offline and the server cannot be reached. If `T` denotes an optional type, the 
  /// method returns a successful status with a value of `nil` for non-existing 
  /// documents.
```

Reply:
Perfect - thanks for the suggestion. Latest commit includes this extended comment. :-)